milli/src/update/index_documents/extract/extract_fid_docid_facet_values.rs

```diff
@@ -21,11 +21,13 @@ use crate::{CboRoaringBitmapCodec, DocumentId, FieldId, Result, BEU32, MAX_FACET
 /// Returns the generated grenad reader containing the docid the fid and the orginal value as key
 /// and the normalized value as value extracted from the given chunk of documents.
 #[logging_timer::time]
+#[allow(clippy::type_complexity)]
```

Comment:
I wonder if we could maybe return an intermediate struct from this function, given that it returns a tuple of many items of the same type, but the order is semantically important.

It would mean changing the return value to `Result<ExtractedFacetValues>`, where:

```rust
pub struct ExtractedFacetValues {
    pub docid_fid_facet_numbers_chunk: grenad::Reader<File>,
    pub docid_fid_facet_strings_chunk: grenad::Reader<File>,
    pub fid_facet_is_null_docids_chunk: grenad::Reader<File>,
    pub fid_facet_exists_docids_chunk: grenad::Reader<File>,
}
```

Reply:
Thank you, fixed, Louis 🏵️